include/pagmo_plugins_nonfree/snopt7.hpp

```diff
@@ -371,11 +378,14 @@ class snopt7 : public not_population_based
      * @param screen_output when ``true`` will activate the screen output from the SNOPT7 library, otherwise
      * will let pagmo regulate logs and screen_output via its pagmo::algorithm::set_verbosity mechanism.
      * @param snopt7_c_library The path to the snopt7_c library.
+     * @param snopt7_minor_version The minor version of your Snopt7 library. Only two APIs are supported at the
+     * moment: 7.6 and 7.7. You may try to use this plugin with different minor version numbers, but at your own risk.
      *
      */
-    snopt7(bool screen_output = false, std::string snopt7_c_library = "/usr/local/lib/libsnopt7_c.so")
-        : m_snopt7_c_library(snopt7_c_library), m_integer_opts(), m_numeric_opts(), m_screen_output(screen_output),
-          m_verbosity(0), m_log(){};
+    snopt7(bool screen_output = false, std::string snopt7_c_library = "/usr/local/lib/libsnopt7_c.so",
+           unsigned minor_version = 6u)
+        : m_snopt7_c_library(snopt7_c_library), m_minor_version(minor_version), m_integer_opts(),
+          m_numeric_opts(), m_screen_output(screen_output), m_verbosity(0), m_log(){};
```

Comment:
I think a check on the ``minor_version`` parameter would be good here.

Reply:
Not sure what to check. Its unsigned. any number > 0 is valid. from 1 to 6 the old API will be triggered, 7 to inf will trigger the other API.

Reply:
I mean that this should work for snopt 7.1 7.2 7.3 7.4 7.5 7.6 with the old API, for snopt 7.7+ with the new API.

If in the future they introduce a further change then we will have to catch it and have a further branching.

Reply:
So it's more like a boolean flag right now? Because phrased like that it seems like it is open to supporting newer api versions in the future, but clearly if the snopt guys put out a new API tomorrow and a user passes in ``minor_version`` 8, it will still crash.

Not sure what is the best course of action to be honest.

Reply:
Right ok I think I see the point better now. This seems like a good place where one would use a warning system, if we had one in pagmo :)

Reply:
The issue I have with enums is that they do not play nicely at the interface with python. Also, since we do not have all the libraries of all versions, nor a description of the API changes is easy to find,  would not guarantee that this works with all the versions.

Its very confusing, I agree. 

I did try to put in the docs warnings on these things. Also about possible crashes if the library version is wrong .... not sure how to make this properly but my guess is that its not possible since snopt7 is developed quite "old fashioned" and does not offer much support for a proper integration.